### PR TITLE
feature - detail order screen

### PR DESCRIPTION
**Pull request by at-duytran**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift [hunk 1]
```diff
@@ -8,9 +8,123 @@
...
+        setupView()
+        tableView.register(DetailOrderTableCell.self)
+
+        // Hud.show()
+        viewModel.fetchOrderDetail { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success:
+                this.updateView()
+                Hud.popActivity()
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift]: cho ra ngoai switch case

File: MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift [hunk 2]
```diff
@@ -8,9 +8,123 @@
...
+            case .failure(let error):
+                this.alert(error: error)
+                Hud.popActivity()
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift]: nt

File: MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift [hunk 3]
```diff
@@ -8,9 +8,123 @@
...
+        orderUpdatedAtLabel.text = String(format: App.Strings.orderUpdatedAtFormat, updatedAt)
+        orderStateNameLabel.text = stateName
+        setupOrderState(stateName: stateName)
+        tableView.reloadData()
+    }
+
+    func setupOrderState(stateName: String) {
+        switch stateName {
+        case App.Strings.stateNew:
+            orderStateNameLabel.textColor = #colorLiteral(red: 0.9372549057, green: 0.3490196168, blue: 0.1921568662, alpha: 1)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/DetailOrder/DetailOrderViewController.swift]: define

File: MyApp/ViewModel/DetailOrder/DetailOrderViewModel.swift [hunk 4]
```diff
@@ -7,3 +7,55 @@
 //
 
 import Foundation
+import MVVM
+
+final class DetailOrderViewModel: MVVM.ViewModel {
+
+    var orderData: DetailOrder?
```
Comment (at-tuanmai2) [inline MyApp/ViewModel/DetailOrder/DetailOrderViewModel.swift]: orderData?

**Pull request merged**